src/frontend/src/eth/utils/qr-code.utils.ts

```diff
@@ -80,7 +94,7 @@ export const decodeQrCode = ({
 	const token =
 		prefix === 'ethereum'
 			? matchEthereumToken({
-					address,
+					tokenAddress: functionName === 'transfer' ? destinationOrTokenAddress : undefined,
```

Comment:
I'm a bit confuse here, is there a use case for a non nullish `destinationOrTokenAddress` with a `functionName === 'transfer'`? or why `undefined` if not transfer?

Reply:
According to ETH standard, the prop that I defined as `destinationOrTokenAddress` is the first address present in the QR code and can be:
- `destination` when the QR code refers to the native token ETH  (mainnet or testnet), so it refers to the destination wallet.
- `tokenAddress` when the QR code refers to an ERC20 token, so it refers to the address of the ERC20 token.

To differentiate the two cases, there is the `functionName` parameter equal to `'transfer'` in the QR code.

So, to check if the QR code refers to an ERC20 token, we check if `functionName` is correct and we pass `destinationOrTokenAddress` to prop `tokenAddress`. If it is referring to the native token, `tokenAddress` prop shall be `undefined` (otherwise we would be incorrectly passing the destination wallet).

Reply:
Thanks for the explanation. Can we document this in some way or add a comment in the code?